phpBB/includes/acp/acp_search.php

```diff
@@ -285,74 +269,36 @@ function index($id, $mode)
 				$this->state[0] = $request->variable('search_type', '');
 			}
 
-			$this->search = null;
-			$error = false;
-			if ($this->init_search($this->state[0], $this->search, $error))
-			{
-				trigger_error($error . adm_back_link($this->u_action), E_USER_WARNING);
-			}
+			$search_backend_factory = $phpbb_container->get('search.backend_factory');
+			$this->search = $search_backend_factory->get($this->state[0]);
```

Comment:
Similar to in the other storage PR, it would be good to change this into using class constants.

Reply:
Ok, in the next pr, but i want to remove state if i can